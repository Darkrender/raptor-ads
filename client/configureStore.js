import { createStore, applyMiddleware, combineReducers } from 'redux';
import createLogger from 'redux-logger';
import rootReducer from './reducers';
import { routerReducer } from 'react-router-redux';

const reducers = combineReducers({
  app: rootReducer,
  routing: routerReducer
})

const configureStore = () => {
  const logger = createLogger();
  return createStore(reducers, applyMiddleware(logger));
}

export default configureStore;
