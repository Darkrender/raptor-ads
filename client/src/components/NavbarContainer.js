import { Navbar } from './Navbar'
import { connect } from 'react-redux'
import { toggleNavigation } from '../actions'

const mapDispatchToProps = (dispatch) => {
  return {
    handleItemClick: (link) => {
      dispatch(toggleNavigation(link));
    }
  }
}

const mapStateToProps = (state) => state

const NavbarContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(Navbar);

export default NavbarContainer;


