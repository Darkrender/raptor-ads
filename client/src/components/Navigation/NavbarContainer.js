import { Navbar } from './Navbar';
import { connect } from 'react-redux';

const mapStateToProps = (state) => {
  const location = state.routing.locationBeforeTransitions.pathname;

  return { location };
}

const NavbarContainer = connect(
  mapStateToProps
)(Navbar);

export default NavbarContainer;


