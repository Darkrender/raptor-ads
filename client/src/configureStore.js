import { createStore, applyMiddleware } from 'redux';
import createLogger from 'redux-logger';
import thunk from 'redux-thunk';
import { routerMiddleware } from 'react-router-redux';
import { browserHistory } from 'react-router';
import rootReducer from './reducers';

const middleware = routerMiddleware(browserHistory);

const configureStore = () => {
  if (process.env.NODE_ENV === 'production') {
    return createStore(rootReducer, applyMiddleware(thunk, middleware));
  }
  const logger = createLogger();
  return createStore(rootReducer, applyMiddleware(logger, thunk, middleware));
};

export default configureStore;
