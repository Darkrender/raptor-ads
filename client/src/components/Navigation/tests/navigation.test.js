import React from 'react';
import NavbarContainer from '../NavbarContainer';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'

const middlewares = []
const mockStore = configureStore(middlewares)
const store = mockStore({routing: {locationBeforeTransitions: {pathname: '/signup'}}});

test('Navigation test', () => {
  //const component = renderer.create(<NavbarContainer store={store}></NavbarContainer>);
  //let tree = component.toJSON();
  //expect(tree).toMatchSnapshot();
})
