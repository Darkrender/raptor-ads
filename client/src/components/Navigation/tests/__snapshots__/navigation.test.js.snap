exports[`test Navigation test 1`] = `
<div
  className="ui inverted segment">
  <div
    className="ui inverted pointing secondary menu">
    <div
      className="header item"
      onClick={[Function]}>
      Our Company
    </div>
    <a
      className="item"
      onClick={[Function]}
      style={Object {}}>
      About
    </a>
    <a
      className="item"
      onClick={[Function]}
      style={Object {}}>
      Customer Dashboard
    </a>
    <div
      className="right menu">
      <a
        className="item"
        onClick={[Function]}
        style={Object {}}>
        Sign Up
      </a>
      <a
        className="item"
        onClick={[Function]}
        style={Object {}}>
        Login
      </a>
    </div>
  </div>
</div>
`;

exports[`test Navigation test 2`] = `
<div
  className="ui inverted segment">
  <div
    className="ui inverted pointing secondary menu">
    <div
      className="header item"
      onClick={[Function]}>
      Our Company
    </div>
    <a
      className="item"
      onClick={[Function]}
      style={Object {}}>
      About
    </a>
    <a
      className="item"
      onClick={[Function]}
      style={Object {}}>
      Customer Dashboard
    </a>
    <div
      className="right menu">
      <a
        className="item"
        onClick={[Function]}
        style={Object {}}>
        Sign Up
      </a>
      <a
        className="item"
        onClick={[Function]}
        style={Object {}}>
        Login
      </a>
    </div>
  </div>
</div>
`;
